src/nnue/nnue_feature_transformer.h

```diff
@@ -420,78 +420,138 @@ namespace Stockfish::Eval::NNUE {
 
       // Now update the accumulators listed in states_to_update[], where the last element is a sentinel.
 #ifdef VECTOR
-      for (IndexType j = 0; j < HalfDimensions / TileHeight; ++j)
+
+      if (   states_to_update[1] == nullptr
+          && (removed[0].size() == 1 || removed[0].size() == 2)
+          && added[0].size() == 1)
       {
-        // Load accumulator
-        auto accTile = reinterpret_cast<vec_t*>(
-          &st->accumulator.accumulation[Perspective][j * TileHeight]);
-        for (IndexType k = 0; k < NumRegs; ++k)
-          acc[k] = vec_load(&accTile[k]);
+          assert(states_to_update[0]);
 
-        for (IndexType i = 0; states_to_update[i]; ++i)
-        {
-          // Difference calculation for the deactivated features
-          for (const auto index : removed[i])
+          auto accTileIn = reinterpret_cast<const vec_t*>(
+              &st->accumulator.accumulation[Perspective][0]);
+          auto accTileOut = reinterpret_cast<vec_t*>(
+              &states_to_update[0]->accumulator.accumulation[Perspective][0]);
+
+          const IndexType offsetR0 = HalfDimensions * removed[0][0];
+          auto columnR0 = reinterpret_cast<const vec_t*>(&weights[offsetR0]);
+          const IndexType offsetA = HalfDimensions * added[0][0];
+          auto columnA = reinterpret_cast<const vec_t*>(&weights[offsetA]);
+
+          if (removed[0].size() == 1)
           {
-            const IndexType offset = HalfDimensions * index + j * TileHeight;
-            auto column = reinterpret_cast<const vec_t*>(&weights[offset]);
-            for (IndexType k = 0; k < NumRegs; ++k)
-              acc[k] = vec_sub_16(acc[k], column[k]);
+              for (IndexType k = 0; k < HalfDimensions * sizeof(std::int16_t) / sizeof(vec_t); ++k)
+                  accTileOut[k] = vec_add_16(vec_sub_16(accTileIn[k], columnR0[k]), columnA[k]);
           }
+          else
+          {
+              const IndexType offsetR1 = HalfDimensions * removed[0][1];
+              auto columnR1 = reinterpret_cast<const vec_t*>(&weights[offsetR1]);
+
+              for (IndexType k = 0; k < HalfDimensions * sizeof(std::int16_t) / sizeof(vec_t); ++k)
+                  accTileOut[k] = vec_add_16(vec_sub_16(vec_sub_16(
```

Comment:
I'd write this as 

```
accTileOut[k] = vec_sub_16(accTileIn[k], columnR0[k]);
accTileOut[k] = vec_sub_16(accTileOut[k], columnR1[k]);
accTileOut[k] = vec_add_16(accTileOut[k], columnA[k]);
```

codegen should be the same. Ditto for the other cases.

I'd also add an assert, and a comment that these operations must produce a dependency chain because otherwise we would run out of registers.

Reply:
@Sopel97  I like your suggestion below to make the operations interdependent.  I think we could do it here also
```
accTileOut[k] = vec_sub_16(
                    vec_add_16(accTileIn[k], columnA[k]),
                    vec_add_16(columnR0[k], columnR1[k]));
```
Do you agree?

Reply:
it should result in a lot of register spilling for AVX2, I could see it being much slower

Reply:
@Sopel97  Ok I tried all three versions on godbolt.  https://godbolt.org/z/fExTPcf31
The original version in this PR and the one w/ the independent operations are the same.  The line by line version you proposed results in more instructions.  Should we then keep it as is or go with the independent operations here and below?  It would be nice to have the psqt code match.   (Disclaimer: I don't really understand register spilling well.)

Reply:
Right, sorry, I understand what's going on now.

> @Sopel97 I like your suggestion below to make the operations interdependent. I think we could do it here also
> 
> ```
> accTileOut[k] = vec_sub_16(
>                     vec_add_16(accTileIn[k], columnA[k]),
>                     vec_add_16(columnR0[k], columnR1[k]));
> ```
> 
> Do you agree?

this is correct, I failed to realize we are no longer bound by the number of registers as the accumulator is not backed by them here. Sorry for the confusion.

Reply:
Ok cool.  So I have updated this and the psqt to the independent operations version and this is ready to be merged now.